packages/number-input/src/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,191 @@
+      value,
+      description,
+      errorMessage,
+      onChange,
+      popoverZIndex,
+      usePortal,
+      portalClassName,
+      portalContainer,
+      scrollContainer,
+      ...rest
+    }: NumberInputProps,
+    forwardedRef,
+  ) => {
+    const prefix = 'lg-numberinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const errorMessageId = useIdAllocator({ prefix, id: ariaDescribedbyProp });
+    const descriptionId = useIdAllocator({ prefix });
+    const selectId = useIdAllocator({ prefix });
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+
+    const hasUnit = !!unit;
+    const hasSelectOptions =
+      Array.isArray(unitOptions) && unitOptions.length > 1;
+
+    const renderUnitOnly = hasUnit && !hasSelectOptions;
+    const renderSelectOnly = hasUnit && hasSelectOptions;
+    const renderErrorMessage = state === State.Error && errorMessage;
+
+    const popoverProps = {
+      popoverZIndex,
+      usePortal,
+      portalClassName,
+      portalContainer,
+      scrollContainer,
+    } as const;
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <div ref={forwardedRef} className={className}>
+          {(label || description) && (
+            <div className={labelDescriptionStyles}>
+              {label && (
+                <Label htmlFor={inputId} disabled={disabled}>
+                  {label}
+                </Label>
+              )}
+              {description && (
+                <Description id={descriptionId} disabled={disabled}>
+                  {description}
+                </Description>
+              )}
+            </div>
+          )}
+          <div
+            className={cx(wrapperBaseStyles, wrapperSizeStyles[size], {
+              [wrapperGapStyles]: renderUnitOnly,
+            })}
+          >
+            <Input
+              value={value}
+              onChange={onChange}
+              disabled={disabled}
+              size={size}
+              id={inputId}
+              hasSelectOptions={renderSelectOnly}
+              state={state}
+              errorMessage={errorMessage}
+              aria-describedby={`${errorMessageId} ${
+                description ? descriptionId : ''
+              } ${renderSelectOnly ? selectId : ''}`}
+              aria-labelledby={
+                !label && ariaLabelledbyProp ? ariaLabelledbyProp : ''
+              }
+              aria-label={!label && ariaLabelProp ? ariaLabelProp : ''}
+              {...rest}
+            />
+            {renderUnitOnly && (
+              <Overline
+                className={cx(unitBaseStyles, unitThemeStyles[theme], {
+                  [unitDisabledStyles[theme]]: disabled,
+                })}
+              >
+                {unit}
+              </Overline>
+            )}
+            {renderSelectOnly && (
+              <Select
+                id={selectId}
+                disabled={disabled}
+                unit={unit as string}
```

Comment:
nit: technically could be undefined (although it shouldn't be), is it safer to default with `""`